src/SWP/Bundle/CoreBundle/Widget/ContentListWidget.php

```diff
@@ -40,13 +39,12 @@ public function render()
         $templateName = $this->getModelParameter('template_name');
         $listId = (int) $this->getModelParameter('list_id');
 
-        /** @var ContentListRepositoryInterface $contentListRepository */
-        $contentListRepository = $this->getContainer()->get('swp.repository.content_list');
         /** @var ContentListInterface $contentList */
-        $contentList = $contentListRepository->findListById($listId);
+        $contentList = $this->getContainer()->get('swp.repository.content_list')->findListById($listId);
 
         return $this->renderTemplate($templateName, [
             'contentList' => $contentList,
```

Comment:
then I guess we dont need to pass the whole content list here, just the required params (listId, limit). Because you are passing now listId but you can actually get it from contentList.